code/game/objects/items/weapons/tankmanipulator.dm

```diff
@@ -0,0 +1,97 @@
+/obj/item/weapon/tankmanip
+	name = "\improper Plasma Tank Manipulator"
+	desc = "A device used by engineering cyborgs to hold and manipulate plasma tanks. Holds up to 3 tanks at a time."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "tankmanip0"
+	opacity = 0
+	density = 0
+	anchored = 0.0
+	var/tanks = 0
+	var/list/storedtanks = list()
+	var/mode = 1
+	w_class = 3.0
+
+/obj/item/weapon/tankmanip/New()
+	desc = "A device used by engineering cyborgs to hold and manipulate plasma tanks. Holds up to 3 tanks at a time."
+	return
+
+/obj/item/weapon/tankmanip/examine()
+	..()
+	usr << "<span class='notice'>It is loaded with [tanks] tanks.</span>"
+
+/obj/item/weapon/tankmanip/update_icon()
+	..()
+	switch(tanks)
+		if(0) icon_state = "tankmanip0"
+		if(1) icon_state = "tankmanip1"
+		if(2) icon_state = "tankmanip2"
+		if(3) icon_state = "tankmanip3"
+
+/obj/item/weapon/tankmanip/attack_self(mob/user)
+	if(tanks)
+		playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 0)
+		while(tanks)
+			var/obj/item/weapon/tank/internals/plasma/T = storedtanks[1]
+			storedtanks.Remove(T)
+			T.loc = usr.loc
+			tanks = storedtanks.len
+		user << "<span class='notice'>You unload the tanks on the floor.</span>"
+		update_icon()
+	return
+
+/obj/item/weapon/tankmanip/afterattack(obj/item/I, mob/user, proximity)
+	..()
+	if (istype(I, /obj/item/weapon/tank/internals/plasma))
+		if (tanks <= 2)
+			playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+			user << "<span class='notice'>You pick up [I].</span>"
+			storedtanks.Add(I)
+			I.loc = src
+			tanks++
+			update_icon()
+			return
+		else
+			user << "<span class='warning'>The manipulator is full!</span>"
+			return
+	//Picks up tanks from the world
+
+/obj/item/weapon/tankmanip/afterattack(obj/machinery/M, mob/user, proximity)
```

Comment:
Is having three afterattack() procs for the same object valid? Does that work? It is not normally combined into one and just having each type check handle its own unique code? Also, there are some common lines in the code here you could combine.